MoleculeVAE-reImplementation/train.py

```diff
@@ -307,13 +322,18 @@ def train(epochs,train_set_final, valid_set_final,logger):
                         torch.save(model.state_dict(), 'Weights/' + str(dt_string.replace("/", '_')) + str(epochs) + '_param.pth')
 
                     print(f'{epochs} / {batch_idx}\t{loss:.4f}')
-        return (train_loss.cpu().detach().numpy() / len(train_set_final)), (valid_loss.cpu().detach().numpy() / len(valid_set_final))
-
-def test(test_set,epochs,logger):
-    dataset=torch.utils.data.DataLoader(test_set, batch_size=params['batch_size'])
+        return (train_loss.cpu().detach().numpy() / len(train_dataloader)), (valid_loss.cpu().detach().numpy() / len(valid_dataloader))
+
+def test(
+        test_set: data.Dataset,
+        epochs,
+        logger,
+        params: Dict,
+        device):
```

Comment:
Since we are now doing the `DataLoader` conversion in this function we need the `params` here to get the `batch_size`.
I've also passed in `device`. It's not clear to me where `device` was coming from in the previous version of the code since it wasn't getting passed in or defined within the namespace.